src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt

```diff
@@ -41,7 +58,14 @@ class CsvFileWriter internal constructor(
      * write rows from Sequence
      */
     override fun writeRows(rows: Sequence<List<Any?>>) {
-        rows.forEach { writeNext(it) }
+        val size = rows.count()
```

Comment:
`Sequence.count` operation is terminal, so all rows are processed on this point.
Kotlin Doc about `Sequence.count` says
> * Returns the number of elements in this sequence.
> * The operation is _terminal_.

Maybe, `rows.iterator()` is better.

Reply:
change at 59297f1, tho not sure if this is what you meant. Please give me your thoughts about this

Reply:
Yes, that's correct

Reply:
Thanks much appreciated.